### ISSUE TITLE
Defective / suboptimal export to StrukTeX

### ISSUE THREAD

**Issue by codemanyak** [fesch/Structorizer.Desktop#497]
The export to StrukTeX creates defective LaTeX documents due to a poor text processing. Moreover, the markup of keywords could be improved (possibly some plugin-specific export options might be added):

- Underscores in the diagram name or its arguments cause typeset errors, they must be escaped.
- String (and character) literals cause typeset errors - the math mode must be switched off around string literals, the quotes must be terminated with an empty brace pair (i.e. `"{}`) lest they should cause trouble as commands.
- Some operator symbols like `^` are to be coded as `^{}` within texts and as `\hat{}` within expressions (math mode) etc.
- Configured keywords like `INPUT`, `OUTPUT`, `for`, `to` etc. should be excluded from math mode and marked up as `\pKey{INPUT}` etc.
- Some element types may also benefit from an update to the most recent StrukTeX version.
- Brackets in the diagram header (e.g. as part of a type specification) must be enclosed in braces, since the diagram header text is put into a command option (otherwise the brackets will disrupt the command option brackets), e.g. `\begin{struktogramm}(119,123)[print\_array(int{[]} values)]`.

**Comment by codemanyak** [comment]
Addition: As there is a opportunity to mark up declarations inside assign-elements with e.g.
```
\begin{declaration}[variable:]
    \description{\pVar{count}}{type \pKey{int} - some comment}
\end{declaration}
```
this should be used for variable declarations and type definitions.
It would also make sense to introduce a declaration element to describe the parameters of a function diagram rather to try to force the argument declarations into the StrukTeX structogram command option (the difficulties of which were indicated in the issue description above).

**Comment by codemanyak** [comment]
The *inparallel* macro of StrukTeX still doesn't cope with directly structured tasks. 
The recent Structorizer export of parallel sections has just concatenated the text of all substructure of a branch into one single line of text, without even switching to math mode. This breaks the LaTeX syntax (and makes not even sense in StrukTeX if put to math mode).
There won't be an alternative to splitting up the diagram into several sub-diagrams, one for each non-atomic branch of the Parallel element to be referenced from the respective *inparallel* task.

**Comment by codemanyak** [comment]
Not a Structorizer but a StrukTeX bug is that the *forever* macro runs havoc (in contrast to a e.g. a *while* macro) if its content exceeds the page width.

**Comment by codemanyak** [comment]
Problems mostly fixed (available on bugfix branch, prepared for 3.27-05)

**Comment by codemanyak** [comment]
W.r.t. the `\return` macro: Unfortunately the macro has two drawbacks - on the one hand, its apparence is not compatible with DIN 66261 (looks like a symmetric exit), on the other hand, the text indentation works diffeerntly from the `\exit` macro (additional spaces must be inserted to achieve a comparable look). So it is questionable whether to use the `\return` macro at all or if we should always cling to the `\exit` macro. Indeed the user might simply replace the command name her-/himself if wanted.
